persistence/trees/trees.go

```diff
@@ -88,6 +98,17 @@ type treeStore struct {
 	treeStoreDir string
 	rootTree     *stateTree
 	merkleTrees  map[string]*stateTree
+
+	// Worldstate holds a previous view of the Worldstate
+	// for savepoints and rollbacks purposes.
+	Prev *Worldstate
```

Comment:
Can you update the name to something more self-explanatory like `PrevState` or `PrevWorlState` or `PrevCommittedState`?